fastify.js

```diff
@@ -794,16 +794,12 @@ function fastify (options) {
       // only call isAsyncConstraint once
       if (isAsync === undefined) isAsync = router.isAsyncConstraint()
       if (rewriteUrl) {
-        const originalUrl = req.url
-        const url = rewriteUrl(req)
-        if (originalUrl !== url) {
-          logger.debug({ originalUrl, url }, 'rewrite url')
-          if (typeof url === 'string') {
-            req.url = url
-          } else {
-            const err = new FST_ERR_ROUTE_REWRITE_NOT_STR(req.url, typeof url)
-            req.destroy(err)
-          }
+        const url = rewriteUrl(req, logger)
```

Comment:
I think it would be better to make the second parameter an object of extra items in order to make future API changes less disruptive:

```suggestion
        const url = rewriteUrl(req, { logger })
```

Reply:
Ok sounds good. Then I will call it `log` so it mimics the shape of `Fastify` (e.g. `fastify.log`)